src/aspire/operators/blk_diag_matrix.py

```diff
@@ -709,6 +728,25 @@ def apply(self, X):
 
         return Y
 
+    def rapply(self, X):
+        """
+        Right apply.  Given a matrix of coefficient vectors,
+        applies the block diagonal matrix on the right hand side.
+        Example, X @ self.
+
+        This is the right hand side equivalent to `apply`.
+
+        :param X: Coefficient matrix, each column is a coefficient vector.
+
+        :return: A matrix with new coefficient vectors.
+        """
+
+        # For now do the transposes a @ b = (b.T @ a.T).T.
+        #  Note there is an optimization opportunity here,
```

Comment:
Is there really? These are all just “logical” transposes in the sense that they just flip a flag in the associated `ndarray`s, correct? Not sure there is much to gain here.

Reply:
Sure. When that flag is flipped it means you are now accessing data with the worst possible strides... Doesn't matter for the small blocks in a block diagonal, but it might for the other dense matrix.

Also, if we go on to use the output matrix.  Is it really stored F or C order....  If it is stored transposed, it might be better to actually transpose it... depending on how large and how many times it is used.

I do agree that there is not much to be gained here today.